theano/tensor/nnet/abstract_conv.py

```diff
@@ -44,9 +44,13 @@ def get_conv_output_shape(image_shape, kernel_shape,
         to: batch size, number of input channels, height and width (and
         possibly depth) of the image. None where undefined.
     kernel_shape: tuple of int (symbolic or numeric) corresponding to the
-        kernel shape. Its four (or five) elements must correspond respectively
-        to: number of output channels, number of input channels, height and
-        width (and possibly depth) of the kernel. None where undefined.
+        kernel shape. For a normal convolution, its four (or five) elements
+        must correspond respectively to : number of output channels, number of
+        input channels, height and width (and possibly depth) of the kernel.
+        For an unshared convolution, its six channels must correspond to :
+        number of output channels, height and width
+        of the output, number of input channels, height and width of the kernel.
+        None where undefined.
```

Comment:
So this mean unshared convolution is only supported for 2d convolution. Can you make that more clear? What about this:

For a normal convolution, its four (2d convolution) or five (3d convolution) ...
For an unshared 2d convolution, ...